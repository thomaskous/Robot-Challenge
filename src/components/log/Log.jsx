import './index.css'
import React from 'react'

/* The purpose of the Log component is to render the log messages 
  (the action history of the robot) by the time the robot was placed on the board.
  Invalid actions are marked with red color */
function Log({ log = [] }) {
  return (
    <div className="log box">
      <div className="log-label">
        Action Log
      </div>
      <div className="log-panel">
        {log.map(({ error, message }, index) => (
          <p
            className={'log-item' + (error ? ' danger' : '')}
            key={index}
            data-testid={index}>
            {message}
          </p>
        ))}
      </div>
    </div>
  );
}

export default Log;