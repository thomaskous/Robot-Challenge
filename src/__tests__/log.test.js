import React from 'react'
import { render } from '@testing-library/react'
import Log from '../components/log/Log'

describe('<Log />', () => {
  it('renders <Log /> with populated data', () => {
    const logs = [
      { error: false, message: 'placing robot on cell X: 0, Y: 0, NORTH' },
      { error: false, message: 'robot rotated right' }
    ];
    const { container, getByText, getByTestId } = render(
      <Log log={logs} />
    );

    expect(getByText('Action Log')).toBeTruthy();
    expect(getByTestId('0')).toBeTruthy();
    expect(getByText('robot rotated right')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});