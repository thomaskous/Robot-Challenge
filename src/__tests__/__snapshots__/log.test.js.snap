// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Log /> renders <Log /> with populated data 1`] = `
<div
  class="log box"
>
  <div
    class="log-label"
  >
    Action Log
  </div>
  <div
    class="log-panel"
  >
    <p
      class="log-item"
      data-testid="0"
    >
      placing robot on cell X: 0, Y: 0, NORTH
    </p>
    <p
      class="log-item"
      data-testid="1"
    >
      robot rotated right
    </p>
  </div>
</div>
`;
